### PR TITLE
Add documentation to editor and files namespaces - closes #2060

### LINKED ISSUE #2060: Add docstrings to editor and files namespaces
As a followup to #2059, there are public fns that still need to be documented for these 2 namespaces. To see fns to be documented, look [at API docs](http://lighttable.github.io/LightTable/api/) and look for "TODO: Add docstring" in namespaces. For editor fns, a lot of them are a direct mirror of CodeMirror fns. It's an open question to decide how we want to document those fns

### PR DESCRIPTION
**Pull request by sbauer322**
This pull request adds documentation to editor and files namespaces. 

One thing to note is that this pull request enables markdown for Codox. This means toggling docs within LightTable will likely cause the markdown syntax to be rendered. A follow up issue #2250 has been opened for future work to address this.

### CODE REVIEW

**Comment by kenny-evitt** [comment]
I am of many minds about this after a brief scan of your changes.

I'm fine with enabling Markdown for Codox.

Linking to the CodeMirror docs is great, except we're using a specific version and you've linked to the always-the-latest-version version of the docs.

I like your 'style' for docstring content but I'd like some buy-in from @rundis and, ideally, @cldwalker too, before we adopt a new convention. If we're not going to adopt a new convention, I don't want to merge changes that 'flout' the one we have now. Your updates to the docstrings are better than the originals, in my opinion, but they're less 'Clojure-y' (based on my experience) and my personal instinct is to just expand the name of the variables, e.g `e` → `editor`, which obviates much of the utility of spelling out what each variable is in the docstring.

I don't want or expect that all of our docstrings be updated all at once but I would like to come to an agreement among the core contributors about what style we want to adopt going forward before we make any changes to existing docstrings and, those less so, add new docstrings with a new style.

**Comment by sbauer322** [comment]
Yes, the CodeMirror links are for the latest but there were only a small handful of differences between the two. I believe we were planning to upgrade to the latest CodeMirror version after getting ClojureScript updated... I know I was anyways. It also served as a low level way of determining how much the API has changed. 

I would prefer the variable names to be more verbose too.

I'm happy to use whichever conventions that are decided upon, but we definitely need at least a link to it for the future. For the editor namespace,  much of it was wrapping the CodeMirror API and was not suited towards examples. For the files namespace, it too was wrapping a good deal, but examples of what was returned seemed beneficial rather than having to test out the functions first...

**Comment by sbauer322** [comment]
I've done some poking around @kenny-evitt and I haven't quite found a idiomatic Clojure docstring resource. Would you have one in mind?

The closest thing I've found is ClojureDocs' [Examples Style Guide](https://clojuredocs.org/examples-styleguide), which I am happy to abide by in the cases where examples are included.

**Comment by rundis** [comment]
- I'm happy to adapt to a more elaborate docs approach that this commit show. 
- A separate issue would be do start gradually start using more meaningful names for params. 
- I'm undecided about the codemirror references though. Quite useful, but will that become a problem when updating versions over time. (Heh well if the api change then that's probably something we need to deal with at the code level, so we might as well deal with upping documentation too then).

**Comment by sbauer322** [comment]
Thanks for the feedback @rundis!

I feel that using meaningful names for params would be beneficial to newcomers looking to contribute (or someone otherwise new to the code in question). Since we don't have much in the way of typing, comments, or docstrings the parameter names are all we have short of looking at source code. In short, I think transitioning to meaningful param names would be useful to improve code readability.

Regarding the codemirror references, like you said, if we already have to deal with an API change then it is trivial to update the documentation at the same time.

**Comment by sbauer322** [comment]
Things to do based on feedback:

- [ ] Reformat examples to use ClojureDocs conventions.
- [ ] Make sure other same namespaced functions linked to in docstrings are properly formatted.
- [ ] Add reference to ClojureDocs example conventions page (and other links on writing quality documentation) somewhere in our documentation - perhaps for-committers.md, or style-guide.md.

File: src/lt/objs/editor.cljs [hunk 1]
```diff
@@ -129,273 +151,512 @@
 ;;*********************************************************
 
 (defn ->val
-  "Return editor's CodeMirror object buffer contents"
+  "Return editor `e`'s buffer contents."
```
Comment (Mouvedia) [inline src/lt/objs/editor.cljs]: content?

File: src/lt/objs/files.cljs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 (ns lt.objs.files
   "Provide fns for doing file related operations. A number of fns
-  use the node fs library - https://nodejs.org/api/fs.html"
+  use the node [fs library](https://nodejs.org/api/fs.html) or [path libarary](https://nodejs.org/api/path.html)."
```
Comment (Mouvedia) [inline src/lt/objs/files.cljs]: library

**Comment by sbauer322** [comment]
Any additional feedback or comments @rundis, @kenny-evitt, or @cldwalker ? I've updated the examples to use ClojureDocs format.

Regarding adopting a new convention for docstrings, I feel it can only better LT to have more in-depth comprehensive documentation (ideally with examples where appropriate). Not having quality API level docs is a huge issue given that being highly configurable is what LT is all about.

I feel the param names should definitely be expanded, but that would be a separate issue.

File: src/lt/objs/files.cljs [hunk 3]
```diff
@@ -77,15 +79,43 @@
            )))
 
 (defn basename
+  "Extracts the basename of the `path`, typically the end of the path.
+
+  If `ext` is provided then the result returned will not contain the extention.
```
Comment (Mouvedia) [inline src/lt/objs/files.cljs]: extension

File: src/lt/objs/files.cljs [hunk 4]
```diff
@@ -94,31 +124,112 @@
         acc
         (recur (rest parts) (conj acc (string/join "." parts)))))))
 
-(defn ext [path]
+(defn ext
+  "Returns the last extention of `path`, without the leading `.`, deteremined by the final `.` of the path.
```
Comment (Mouvedia) [inline src/lt/objs/files.cljs]: determined

File: src/lt/objs/files.cljs [hunk 5]
```diff
@@ -94,31 +124,112 @@
...
+  (ext \"foo.\")            ;;=> \"\"
+
+  (ext \"foo\")             ;;=> \"\"
+  ```"
+  [path]
   (subs (.extname fpath path) 1))
 
-(defn without-ext [path]
+(defn without-ext
+  "Returns the `path`, but without the last extension, deteremined by the final `.` of the path.
```
Comment (Mouvedia) [inline src/lt/objs/files.cljs]: again

**Comment by rundis** [comment]
Excellent work @sbauer322 . Tx so much for this, really useful.
Have a look at the comments from @Mouvedia , once done squash the commits, rebase and merge :-)

File: src/lt/objs/editor.cljs [hunk 6]
```diff
@@ -1,7 +1,21 @@
 (ns lt.objs.editor
   "Provide fns and behaviors for interfacing with a CodeMirror editor
-  object. Also manage defining and loading CodeMirror. For more about CodeMirror
-  objects see http://codemirror.org/doc/manual.html#CodeMirror"
+  object. Also manage defining and loading [CodeMirror](http://codemirror.net/doc/manual.html).
+
+  Editor objects are frequently used as arguements for functions, but often only the internal
```
Comment (Mouvedia) [inline src/lt/objs/editor.cljs]: arguments

File: src/lt/objs/editor.cljs [hunk 7]
```diff
@@ -1,7 +1,21 @@
...
+  CodeMirror object is actually used. Where the following documentation referers to the editor,
+  it is informally referring to the editor's CodeMirror object.
+
+  Commonly encountered arguement names:
```
Comment (Mouvedia) [inline src/lt/objs/editor.cljs]: argument

**Pull request merged**